demisto_sdk/commands/download/downloader.py

```diff
@@ -439,305 +460,397 @@ def build_req_params(self):
-                )
-                if content_object:
-                    self.pack_content[content_entity].append(content_object)
+
+        # All validations passed
+        else:
+            return True
+
+        return False
+
+    def build_existing_pack_structure(self, existing_pack_path: Path) -> dict[str, dict[str, list[dict]]]:
+        """
+        Create a pack structure from the content in the existing output path.
+        Used later to determine which files already exist in the output path (by their content name, not file name).
+
+        Args:
+            existing_pack_path (Path): The path of the existing pack to parse.
+
+        Returns:
+            dict[str, dict[str, list[dict]]]: A dictionary representing the pack structure.
+
+        Example return structure:
+        {
+            "Integrations":
+                "MyIntegration":
+                    [
+                        {
+                            "name": "MyIntegration",
+                            "id": "MyIntegration",
+                            "path": "Packs/TestPack/Integrations/MyIntegration/MyIntegration.yml",
+                            "file_extension": "yml"
+                        },
+                        {
+                            "name": "MyIntegration",
+                            "id": "MyIntegration",
+                            "path": "Packs/TestPack/Integrations/MyIntegration/MyIntegration.py",
+                            "file_extension": "py"
+                        },
+                        ... (more files, like README and description)
+                    ]
+        """
+        logger.info(f"Parsing existing content in '{existing_pack_path}'...")
+        pack_structure: DefaultDict[str, dict[str, list]] = defaultdict(dict)
+
+        for content_entity_path in existing_pack_path.iterdir():
+            if content_entity_path.is_dir():
+                directory_name = content_entity_path.name
+
+                if directory_name in (INTEGRATIONS_DIR, SCRIPTS_DIR):
+                    # If entity is of type integration/script it will have dirs, otherwise files
+                    entity_instances_paths = [p for p in content_entity_path.iterdir() if p.is_dir()]
+                else:
+                    entity_instances_paths = [p for p in content_entity_path.iterdir() if p.is_file()]
+
+                for entity_instance_path in entity_instances_paths:
+                    content_data = self.build_pack_content_object(
+                        content_entity=directory_name,
+                        entity_instance_path=entity_instance_path
+                    )
+
+                    if content_data is not None:
+                        content_name, content_object = content_data
+                        pack_structure[directory_name][content_name] = content_object
+
+        logger.info("Parsing of existing content items completed.")
+        return dict(pack_structure)
```

Comment:
It is, it's a subclass (with some additional functionality).
But since I'm returning it for usage in multiple other functions, I think it will make the code easier to maintain for others in the future if it's just a normal dict (the defaultdict-specific functionality is no longer relevant after the creation in this method).